### PR TITLE
Feature/support readonlyrootfilesystem config

### PR DESCRIPTION
**Pull request by LucasRoesler**
Ports the changes from openfaas/faas-netes#248 to openfaas-operator.  This adds support for the `ReadOnlyRootFilesystem` flag on the function definition, in particular, this will:

- Set the ReadOnlyRootFilesystem option in the SecurityContext of the
function
- Add a writable mount at `/tmp` when the option is true
- Remove the read only configurations when the option is false
- Add helper methods to determine when the read-only configuration has
changed
- Bumpe the version of `faas`

Note that this change depends on another PR #33 to in which the vendor folder is pruned.

### CODE REVIEW

File: pkg/apis/openfaas/v1alpha2/types.go [hunk 1]
```diff
@@ -19,16 +19,17 @@ type Function struct {
...
+	Image                  string             `json:"image"`
+	Replicas               *int32             `json:"replicas"`
+	Handler                string             `json:"handler"`
+	Labels                 *map[string]string `json:"labels"`
+	Environment            *map[string]string `json:"environment"`
+	Constraints            []string           `json:"constraints"`
+	Secrets                []string           `json:"secrets"`
+	Limits                 *FunctionResources `json:"limits"`
+	Requests               *FunctionResources `json:"requests"`
+	ReadOnlyRootFilesystem bool               `json:"readOnlyRootFilesystem"`
```
Comment (stefanprodan) [inline pkg/apis/openfaas/v1alpha2/types.go]: @LucasRoesler has you tested this with functions deployed based on the old CRD?

**Comment by alexellis** [comment]
In addition to Stefan's request I wanted to let you both know that I'm seeing a conflict on: Gopkg.lock.

**Comment by alexellis** [comment]
@LucasRoesler please could you comment on Stefan's request when you have time?

**Comment by alexellis** [comment]
Merge blocked awaiting further response.

**Comment by LucasRoesler** [comment]
I tried this a couple times tonight without any errors.  I installed everything from master, installed some functions, and then updated them to toggle the readOnly flag a couple times.

I am noticing that the tests are failing on travis, but I am not seeing any of the issues locally.

**Comment by stefanprodan** [comment]
@LucasRoesler use ./hack/update-codegen.sh to update the generated code with the new field and commit the changes. The deep copy in your branch is missing the `ReadOnlyRootFilesystem` filed https://github.com/LucasRoesler/openfaas-operator/blob/5dd0a95b6489437310c59cd036cb9b758230d6ef/pkg/apis/openfaas/v1alpha2/zz_generated.deepcopy.go#L95

**Comment by LucasRoesler** [comment]
@stefanprodan I do not have any diff to commit when i run `update-codegen`.  I have committed an pushed all of my changes so far, including any generated changes

**Comment by stefanprodan** [comment]
@LucasRoesler you are right, the deep copy will not change since ReadOnlyRootFilesystem is of type boolean and it doesn't need to be reflected. I see that the latest commit was successfully built in Travis.

**Comment by alexellis** [comment]
Merging, thank you @LucasRoesler 

How did you find working with the operator to make a change? Any tips or alterations we can make in the README?

**Comment by alexellis** [comment]
@edwilde this may be a good example to follow for the annotations merge.

**Comment by ewilde** [comment]
@alexellis the wrong `Ed Wilde` :smile: i'm `@ewilde`

**Comment by alexellis** [comment]
Oops, sorry Eds.

**Comment by LucasRoesler** [comment]
@alexellis no major changes to recommend.  I used the helm based install, which required modifying the helm chart, but I don't think anything unusual

**Pull request merged**